src/JsonRpc/JsonRpcServerServiceCollectionExtensions.cs

```diff
@@ -79,23 +81,51 @@ internal static IContainer AddJsonRpcMediatR(this IContainer container)
             container.RegisterMany(new[] { typeof(IMediator).GetAssembly() }, Registrator.Interfaces, Reuse.ScopedOrSingleton);
             container.RegisterMany<RequestContext>(Reuse.Scoped);
             container.RegisterDelegate<ServiceFactory>(context => context.Resolve, Reuse.ScopedOrSingleton);
+            container.Register(typeof(IRequestHandler<,>), typeof(RequestHandler<,>));
+            container.Register(typeof(IRequestHandler<,>), typeof(RequestHandlerDecorator<,>), setup: Setup.Decorator);
+
+            return container;
+        }
+
+        class RequestHandler<T, R> : IRequestHandler<T, R> where T : IRequest<R>
+        {
+            private readonly IRequestContext _requestContext;
+
+            public RequestHandler(IRequestContext requestContext)
+            {
+                _requestContext = requestContext;
+            }
+            public Task<R> Handle(T request, CancellationToken cancellationToken)
+            {
+                return ((IRequestHandler<T, R>) _requestContext.Descriptor.Handler).Handle(request, cancellationToken);
+            }
+        }
+
+        class RequestHandlerDecorator<T, R> : IRequestHandler<T, R> where T : IRequest<R>
+        {
+            private readonly IRequestHandler<T, R>? _handler;
+            private readonly IRequestContext? _requestContext;
+
+            public RequestHandlerDecorator(IRequestHandler<T, R>? handler = null, IRequestContext? requestContext = null)
+            {
+                _handler = handler;
+                _requestContext = requestContext;
+            }
+            public Task<R> Handle(T request, CancellationToken cancellationToken)
+            {
+                if (_requestContext == null)
+                {
+                    if (_handler == null)
+                    {
+                        throw new NotImplementedException($"No request handler was registered for type {typeof(IRequestHandler<T, R>).FullName}");
 
-            return container.With(
-                rules => rules.WithUnknownServiceResolvers(
```

Comment:
The fundamental problem was, the "built-in" handlers are getting added to the container, so they are never "unknown", and therefore this code doesn't run.